File: src/java/PriceManager/CurrencyConverter.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package PriceManager;

/**
 *
 * @author Paul Sweeney
 */
public class CurrencyConverter {
    
    public double convert(double originalAmount, String newCurrencyCountry) {
        double newAmount = originalAmount;
        
        if (newCurrencyCountry.equals("IE") || newCurrencyCountry.equals("FR")) {
            newAmount = originalAmount;
        } else if (newCurrencyCountry.equals("GB")) {
            newAmount = originalAmount * 0.87;
        } else if (newCurrencyCountry.equals("US")) {
            newAmount = originalAmount * 1.13;
        } else if (newCurrencyCountry.equals("en-CA") || newCurrencyCountry.equals("fr-CA")) {
            newAmount = originalAmount * 1.50;
        } else {
            newAmount = -1;
        }
        
        return newAmount;
    }
    
}

File: web/Javascript/currencyConverter.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


var https = require('https');

// Creating currency converter object
var currencyConverter = new Object();

currencyConverter.convert = function convertCurrency(amount, fromCurrency, toCurrency, cb) {
  var apiKey = '182238e3f927a2d5a2b3';

  fromCurrency = encodeURIComponent(fromCurrency);
  toCurrency = encodeURIComponent(toCurrency);
  var query = fromCurrency + '_' + toCurrency;

  var url = 'https://api.currencyconverterapi.com/api/v6/convert?q='
            + query + '&compact=ultra&apiKey=' + apiKey;

  https.get(url, function(res){
      var body = '';

      res.on('data', function(chunk){
          body += chunk;
      });

      res.on('end', function(){
          try {
            var jsonObj = JSON.parse(body);

            var val = jsonObj[query];
            if (val) {
              var total = val * amount;
              cb(null, Math.round(total * 100) / 100);
            } else {
              var err = new Error("Value not found for " + query);
              console.log(err);
              cb(err);
            }
          } catch(e) {
            console.log("Parse error: ", e);
            cb(e);
          }
      });
  }).on('error', function(e){
        console.log("Got an error: ", e);
        cb(e);
  });
};

//uncomment to test

convertCurrency(10, 'USD', 'PHP', function(err, amount) {
  console.log(amount);
});